bagua/torch_api/algorithms/decentralized.py

```diff
@@ -99,27 +101,29 @@ def hook(parameter_name, parameter):
 
     def init_post_backward_hook(self, bagua_module: BaguaModule):
         def hook():
-            for bucket in bagua_module.bagua_buckets:
-                for tensor in bucket.tensors:
-                    tensor.bagua_mark_communication_ready()
-
-            bagua_module._bagua_backend.wait_pending_comm_ops()
+            pass
 
         return hook
 
     def init_post_optimizer_step_hook(self, bagua_module: BaguaModule):
         def hook(optimizer: torch.optim.Optimizer):
-            bagua_module._bagua_backend.execute_post_optimizer_step_comm_ops()
-            bagua_module._bagua_backend.wait_pending_post_optimizer_step_comm_ops()
+            for bucket in bagua_module.bagua_buckets:
+
+                for tensor in bucket.tensors:
+                    tensor.bagua_mark_communication_ready()
+
+            bagua_module._bagua_backend.wait_pending_comm_ops()
 
         return hook
 
     def _init_states(self, bucket: BaguaBucket):
         bucket_flattened_tensor = bucket.flattened_tensor()
 
+        weight_tensor = bucket_flattened_tensor.detach().clone()
         left_peer_weight_tensor = bucket_flattened_tensor.detach().clone()
         right_peer_weight_tensor = bucket_flattened_tensor.detach().clone()
 
+        bucket.set_state("weight", weight_tensor)
```

Comment:
don't use dictionary operation like `set_state(str, object)`, just do bucket.whatever_you_want = object

dictionary operation breaks static analysis and intellisense